tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -280,6 +284,89 @@ internal void UpdateAddedProbeInstrumentations(IReadOnlyList<ProbeDefinition> ad
             }
         }
 
+        private static string[] SafeParseSignature(string signature)
+        {
+            if (string.IsNullOrEmpty(signature))
+            {
+                return null;
+            }
+
+            try
+            {
+                return ParseSignature(signature);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Error parsing the signature {Signature}. Fall back to empty signature", signature);
+                return null;
+            }
+        }
+
+        private static string[] ParseSignature(string signature)
+        {
+            if (signature.StartsWith("(") && signature.EndsWith(")"))
+            {
+                signature = signature.Substring(1, signature.Length - 2);
+            }
+
+            var types = new List<string>();
+            var initialCapacity = 32;
+            var currentType = StringBuilderCache.Acquire(initialCapacity);
+            var stack = new Stack<char>();
+
+            try
+            {
+                foreach (char c in signature)
+                {
+                    if (c == ' ')
+                    {
+                        continue;
+                    }
+
+                    if (c == ',' && stack.Count == 0)
+                    {
+                        types.Add(StringBuilderCache.GetStringAndRelease(currentType));
+                        currentType = StringBuilderCache.Acquire(initialCapacity);
+                    }
+                    else
+                    {
+                        if (c == '<')
+                        {
+                            stack.Push(c);
+                            currentType.Append('['); // Replace < with [
+                        }
+                        else if (c == '>')
+                        {
+                            if (stack.Count > 0 && stack.Peek() == '<')
+                            {
+                                stack.Pop();
+                                currentType.Append(']'); // Replace > with ]
+                            }
+                            else
+                            {
+                                throw new InvalidOperationException("Mismatched angle brackets in the signature.");
+                            }
+                        }
+                        else
+                        {
+                            currentType.Append(c);
+                        }
+                    }
+                }
+
+                if (currentType.Length > 0)
+                {
+                    types.Add(StringBuilderCache.GetStringAndRelease(currentType));
+                }
+            }
+            finally
+            {
+                StringBuilderCache.Release(currentType);
+            }
+
+            return types.ToArray();
+        }
```

Comment:
Do we have unit tests for that? IMO we must add some to verify the correctness because it's in a very critical path (adding probes).

Some suggestions:

- `P<T<V,K>> arg1, K<T<V>, P<K>>arg2`
- `<c>_someGeneratedType arg1, anotherOne<>c arg2`
- Also worth checking arguments with attributes and default values

Reply:
For the more bizarre signatures we shall first assess what the SymDb is uploading. I'm currently adding unit tests for the signature parser, including generics.

Reply:
Added the unit tests in d09b0d31c3e24c41453e141423608387754c5534.